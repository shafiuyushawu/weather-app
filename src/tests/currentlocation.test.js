import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import CurrentCondition from '../components/CurrentCondtion';
import '@testing-library/jest-dom/extend-expect';

const mockStore = configureStore([]);

describe('Check if current location components render correctly', () => {
  test('renders CurrentCondition component', () => {
    const store = mockStore({
      weather: {
        conditions: {},
      },
    });

    render(
      <Provider store={store}>
        <CurrentCondition />
      </Provider>,
    );

    expect(screen.getByText('Current conditions')).toBeInTheDocument();
    expect(screen.getByText('Last Updated:')).toBeInTheDocument();
  });

  test('renders with conditions', () => {
    const store = mockStore({
      weather: {
        conditions: {
          last_updated: '2022-04-20 12:00',
          temp_c: '20',
          humidity: '70',
          cloud: '50',
          wind_mph: '10',
          icon: 'https://example.com/icon.png',
          text: 'Partly cloudy',
        },
      },
    });

    const { container } = render(
      <Provider store={store}>
        <CurrentCondition />
      </Provider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders CurrentCondition component', () => {
    const store = mockStore({
      weather: {
        conditions: {},
      },
    });

    render(
      <Provider store={store}>
        <CurrentCondition />
      </Provider>,
    );

    expect(screen.getByText('Current conditions')).toBeInTheDocument();
    expect(screen.getByText('Last Updated:')).toBeInTheDocument();
  });
});
