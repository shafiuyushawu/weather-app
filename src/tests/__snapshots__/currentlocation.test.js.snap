// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if current location components render correctly renders with conditions 1`] = `
<div
  class=""
>
  <h1
    class="text-xl bg-gradient-to-tr from-blue-800 to-green-600 text-white flex justify-center items-center gap-3 mb-5"
  >
    Current conditions
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="4"
      />
      <path
        d="M13 4.069V2h-2v2.069A8.01 8.01 0 0 0 4.069 11H2v2h2.069A8.008 8.008 0 0 0 11 19.931V22h2v-2.069A8.007 8.007 0 0 0 19.931 13H22v-2h-2.069A8.008 8.008 0 0 0 13 4.069zM12 18c-3.309 0-6-2.691-6-6s2.691-6 6-6 6 2.691 6 6-2.691 6-6 6z"
      />
    </svg>
  </h1>
  <div
    class="flex flex-col"
  >
    <h2
      class="text-center "
    >
      Last Updated:
         
      <span
        class="text-white"
      >
        <p>
          2022-04-20 12:00
        </p>
      </span>
    </h2>
    <div
      class="flex justify-between gap-5"
    >
      <ul
        class="grid grid-cols-1 md:grid-cols-1 xl:grid-cols-4 mb-5"
      >
        <li
          class="flex"
        >
          Temperature:
           
          <span>
            <p>
              20
            </p>
          </span>
          <svg
            class="text-2xl text-white"
            fill="currentColor"
            height="1em"
            id="Layer_1"
            stroke="currentColor"
            stroke-width="0"
            version="1.1"
            viewBox="0 0 30 30"
            width="1em"
            x="0px"
            xmlns="http://www.w3.org/2000/svg"
            y="0px"
          >
            <path
              d="M13.19,9.21c0-0.5,0.18-0.93,0.53-1.28c0.36-0.36,0.78-0.53,1.28-0.53c0.49,0,0.92,0.18,1.27,0.53
	c0.35,0.36,0.53,0.78,0.53,1.28s-0.18,0.93-0.53,1.29c-0.35,0.36-0.78,0.54-1.27,0.54c-0.49,0-0.92-0.18-1.28-0.54
	S13.19,9.71,13.19,9.21z M14.07,9.21c0,0.26,0.09,0.48,0.27,0.67c0.19,0.19,0.41,0.28,0.67,0.28c0.26,0,0.48-0.09,0.67-0.28
	s0.28-0.41,0.28-0.67c0-0.26-0.09-0.48-0.28-0.66c-0.19-0.18-0.41-0.28-0.67-0.28c-0.26,0-0.48,0.09-0.67,0.27
	C14.16,8.72,14.07,8.94,14.07,9.21z"
            />
          </svg>
        </li>
        <li
          class="flex"
        >
          Humidity:
           
          <p>
            70
          </p>
          <svg
            fill="currentColor"
            height="1em"
            id="Layer_1"
            stroke="currentColor"
            stroke-width="0"
            version="1.1"
            viewBox="0 0 30 30"
            width="1em"
            x="0px"
            xmlns="http://www.w3.org/2000/svg"
            y="0px"
          >
            <path
              d="M7.56,17.19c0-0.88,0.24-1.89,0.72-3.03s1.1-2.25,1.86-3.31c1.56-2.06,2.92-3.62,4.06-4.67l0.75-0.72
	c0.25,0.26,0.53,0.5,0.83,0.72c0.41,0.42,1.04,1.11,1.88,2.09s1.57,1.85,2.17,2.65c0.71,1.01,1.32,2.1,1.81,3.25
	s0.74,2.16,0.74,3.03c0,1-0.19,1.95-0.58,2.86c-0.39,0.91-0.91,1.7-1.57,2.36c-0.66,0.66-1.45,1.19-2.37,1.58
	c-0.92,0.39-1.89,0.59-2.91,0.59c-1,0-1.95-0.19-2.86-0.57c-0.91-0.38-1.7-0.89-2.36-1.55c-0.66-0.65-1.19-1.44-1.58-2.35
	S7.56,18.23,7.56,17.19z M9.82,14.26c0,0.83,0.17,1.49,0.52,1.99c0.35,0.49,0.88,0.74,1.59,0.74c0.72,0,1.25-0.25,1.61-0.74
	c0.35-0.49,0.53-1.15,0.54-1.99c-0.01-0.84-0.19-1.5-0.54-2c-0.35-0.49-0.89-0.74-1.61-0.74c-0.71,0-1.24,0.25-1.59,0.74
	C9.99,12.76,9.82,13.42,9.82,14.26z M11.39,14.26c0-0.15,0-0.27,0-0.35s0.01-0.19,0.02-0.33c0.01-0.14,0.02-0.25,0.05-0.32
	s0.05-0.16,0.09-0.24c0.04-0.08,0.09-0.15,0.15-0.18c0.07-0.04,0.14-0.06,0.23-0.06c0.14,0,0.25,0.04,0.33,0.12s0.14,0.21,0.17,0.38
	c0.03,0.18,0.05,0.32,0.06,0.45s0.01,0.3,0.01,0.52c0,0.23,0,0.4-0.01,0.52c-0.01,0.12-0.03,0.27-0.06,0.45
	c-0.03,0.17-0.09,0.3-0.17,0.38s-0.19,0.12-0.33,0.12c-0.09,0-0.16-0.02-0.23-0.06c-0.07-0.04-0.12-0.1-0.15-0.18
	c-0.04-0.08-0.07-0.17-0.09-0.24c-0.02-0.08-0.04-0.19-0.05-0.32c-0.01-0.14-0.02-0.25-0.02-0.32S11.39,14.41,11.39,14.26z
	 M11.98,22.01h1.32l4.99-10.74h-1.35L11.98,22.01z M16.28,19.02c0.01,0.84,0.2,1.5,0.55,2c0.35,0.49,0.89,0.74,1.6,0.74
	c0.72,0,1.25-0.25,1.6-0.74c0.35-0.49,0.52-1.16,0.53-2c-0.01-0.84-0.18-1.5-0.53-1.99c-0.35-0.49-0.88-0.74-1.6-0.74
	c-0.71,0-1.25,0.25-1.6,0.74C16.47,17.52,16.29,18.18,16.28,19.02z M17.85,19.02c0-0.23,0-0.4,0.01-0.52
	c0.01-0.12,0.03-0.27,0.06-0.45s0.09-0.3,0.17-0.38s0.19-0.12,0.33-0.12c0.09,0,0.17,0.02,0.24,0.06c0.07,0.04,0.12,0.1,0.16,0.19
	c0.04,0.09,0.07,0.17,0.1,0.24s0.04,0.18,0.05,0.32l0.01,0.32l0,0.34c0,0.16,0,0.28,0,0.35l-0.01,0.32l-0.05,0.32l-0.1,0.24
	l-0.16,0.19l-0.24,0.06c-0.14,0-0.25-0.04-0.33-0.12s-0.14-0.21-0.17-0.38c-0.03-0.18-0.05-0.33-0.06-0.45S17.85,19.25,17.85,19.02z
	"
            />
          </svg>
        </li>
        <li
          class="flex"
        >
          Cloud:
           
          <p>
            50
          </p>
          <svg
            aria-hidden="true"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.657 16.657L13.414 20.9a1.998 1.998 0 01-2.827 0l-4.244-4.243a8 8 0 1111.314 0z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
            <path
              d="M15 11a3 3 0 11-6 0 3 3 0 016 0z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </li>
        <li
          class="flex"
        >
          Wind:
           
          <p>
            10
          </p>
          mph
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M272 16c-32 48-38.97 115.99-38 176 .118 7.3 2.486 17.54 5.086 26.69 5.166-2.36 10.89-3.69 16.914-3.69 3.04 0 6 .346 8.857.982L272 16zm-16 217c-12.81 0-23 10.19-23 23s10.19 23 23 23 23-10.19 23-23-10.19-23-23-23zm40.8 27.033c-.897 9.054-4.776 17.268-10.632 23.643L455.812 389.87c-25.57-51.714-80.964-91.744-133.42-120.91-6.366-3.54-16.386-6.605-25.593-8.927zm-79.896 8.274L40.124 362.12c57.57 3.714 119.937-24.243 171.423-55.09 6.274-3.758 13.95-10.936 20.58-17.77-7.1-5.122-12.522-12.447-15.223-20.953zm55.86 25.07c-5.127 2.32-10.8 3.623-16.764 3.623-5.964 0-11.637-1.302-16.764-3.62L232 496h48l-7.236-202.623z"
            />
          </svg>
        </li>
      </ul>
      <div>
        <img
          alt="icon"
          src="https://example.com/icon.png"
        />
        <span>
          <p>
            Partly cloudy
          </p>
        </span>
      </div>
    </div>
  </div>
</div>
`;
